### PR TITLE
Refactored all site config default hardcoded value in rspec test

### PR DESCRIPTION
**Pull request by shilpi230**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
- Refactored all site config default harcoded value to use site config default values.

## Related Tickets & Documents
Closes [#12283](https://github.com/forem/forem/issues/12283)

## Added tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [ ] README
- [x] No documentation needed

### CODE REVIEW

**Comment by shilpi230** [comment]
@mstruve I raised the PR but some of the test case failed. I wonder how ? I didn't change anything in code. Just one test case file I changed. Can you please help me with it. Thanks!

**Comment by mstruve** [comment]
Once this is merged https://github.com/forem/forem/pull/12313 try updating your branch, that should fix it

File: spec/requests/admin/configs_spec.rb [hunk 1]
```diff
@@ -347,7 +347,7 @@
         end
 
         it "updates logo_png" do
-          expected_default_image_url = URL.local_image("icon.png")
+          expected_default_image_url = SiteConfig.get_field(:logo_png).dig(:default)
```
Comment (citizen428) [inline spec/requests/admin/configs_spec.rb:350]: How do you feel about adding this as a method to `SiteConfig` itself?

```ruby
def self.get_default(field)
  get_field(field).dig(:default).call
end
```

Then we could just use `SiteConfig.get_default(:logo_png)`, etc. here in the spec.

**Review by citizen428 — APPROVED** [review]

File: spec/requests/admin/configs_spec.rb (hunk 1 above)
Comment (shilpi230) [reply spec/requests/admin/configs_spec.rb:350]: Makes sense @citizen428 looks more neater. Thank you!

**Comment by rhymes** [comment]
@shilpi230 I guess you had that conflict because you cherry picked the commit instead of merging master, correct? Not a big deal as we squash changes but a FYI for the next time: better to merge master in or rebase :)

**Comment by shilpi230** [comment]
No @rhymes I didn't cherry picked. I rebased it, but it wasn't allowing me to push the code. And I read it in some of the code of conduct that its advisable not force push when your branch is under review. Thats why I had to pull and push the changes, at that time I got the conflict, within my own code. I can still squash and push the code. Let me know. Thanks!

**Comment by shilpi230** [comment]
One of the [travis](https://travis-ci.com/github/forem/forem/jobs/472598030) build failed. Can someone please help with it. Thank you!

**Comment by rhymes** [comment]
Ah got it @shilpi230, I restarted the build, you can check it here https://travis-ci.com/github/forem/forem/jobs/472598030

**Comment by shilpi230** [comment]
Thank you! @rhymes

**Review by rhymes — APPROVED** [review]
LGTM.
Thank you @shilpi230!

**Pull request merged**